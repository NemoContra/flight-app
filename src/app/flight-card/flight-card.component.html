<div [ngClass]="{ selected: selected() }" class="card">
  <div class="card-header">
    <h2 class="title">
      {{ item().from | city: "short" }} - {{ item().to | city: "long" }}
    </h2>
  </div>

  <div class="card-body">
    <p>Flight-No.: #{{ item().id }}</p>
    <p>Date: {{ item().date | date: "dd.MM.yyyy HH:mm" }}</p>
    <p [ngStyle]="{ color: item().delayed ? 'red' : null }">
      Delayed: {{ item().delayed }}
    </p>
    <p>
      <button (click)="toggleSelected()" class="btn btn-default">
        {{ selected() ? "Remove" : "Select" }}</button
      ><button (click)="edit()" class="btn btn-default">Edit</button>
    </p>
  </div>
</div>
