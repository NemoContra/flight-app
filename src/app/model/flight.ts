export abstract class Flight {
  abstract id: number;
  abstract from: string;
  abstract to: string;
  abstract date: string;
  abstract delayed: boolean;
}
