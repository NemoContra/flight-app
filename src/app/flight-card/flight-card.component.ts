import {
  ChangeDetectionStrategy,
  Component,
  inject,
  input,
  model,
} from '@angular/core';
import { DatePipe, NgClass, NgStyle } from '@angular/common';
import { Flight } from '../model/flight';
import { CityPipe } from '../shared/city.pipe';
import { MatDialog } from '@angular/material/dialog';
import { FlightEditComponent } from '../flight-edit/flight-edit.component';
import { FormBuilder } from '@angular/forms';

@Component({
  selector: 'app-flight-card',
  standalone: true,
  imports: [CityPipe, NgClass, NgStyle, DatePipe],
  templateUrl: './flight-card.component.html',
  styleUrls: ['./flight-card.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class FlightCardComponent {
  item = input.required<Flight>();
  selected = model<boolean>(false);

  toggleSelected() {
    this.selected.update((selected) => !selected);
  }

  dialog = inject(MatDialog);

  edit() {
    this.dialog.open(FlightEditComponent, {
      data: { flight: this.item() },
    });
  }
}
