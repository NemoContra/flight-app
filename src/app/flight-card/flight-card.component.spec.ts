import { createHostFactory } from '@ngneat/spectator/jest';
import { FlightCardComponent } from './flight-card.component';
import { Flight } from '../model/flight';

describe('FlightCardComponent', () => {
  const createComponent = createHostFactory(FlightCardComponent);

  it('should render a card', () => {
    const spectator = createComponent(
      `<app-flight-card selected="false" [item]="item">`,
      {
        hostProps: {
          item: {
            id: 1,
            from: 'London',
            to: 'Paris',
            date: '2021-12-24T12:00:00Z',
            delayed: false,
          } satisfies Flight,
        },
      },
    );

    expect(spectator.fixture).toMatchSnapshot();
  });
});
