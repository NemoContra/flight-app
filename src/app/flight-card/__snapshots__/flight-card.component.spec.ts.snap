// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightCardComponent should render a card 1`] = `
<lib-ngneat-host-component
  item={[Function Object]}
>
  <app-flight-card
    selected="false"
  >
    <div
      class="card selected"
    >
      <div
        class="card-header"
      >
        <h2
          class="title"
        >
           LCY - Charles de Gaulle Airport 
        </h2>
      </div>
      <div
        class="card-body"
      >
        <p>
          Flight-No.: #1
        </p>
        <p>
          Date: 24.12.2021 13:00
        </p>
        <p>
           Delayed: false 
        </p>
        <p>
          <button
            class="btn btn-default"
          >
             Remove
          </button>
          <button
            class="btn btn-default"
          >
            Edit
          </button>
        </p>
      </div>
    </div>
  </app-flight-card>
</lib-ngneat-host-component>
`;
